applications/RomApplication/custom_strategies/rom_builder_and_solver.h

```diff
@@ -160,35 +160,52 @@ class ROMBuilderAndSolver : public BuilderAndSolver<TSparseSpace, TDenseSpace, T
 
         #pragma omp parallel firstprivate(dof_list, second_dof_list)
         {
-            auto &r_current_process_info = rModelPart.GetProcessInfo();
+            const ProcessInfo& r_current_process_info = rModelPart.GetProcessInfo();
 
             // We cleate the temporal set and we reserve some space on them
             set_type dofs_tmp_set;
             dofs_tmp_set.reserve(20000);
-
             // Gets the array of elements from the modeler
             #pragma omp for schedule(guided, 512) nowait
             for (int i = 0; i < number_of_elements; ++i)
             {
                 auto it_elem = r_elements_array.begin() + i;
-
+                //detect whether the element has a Hyperreduced Weight (H-ROM simulation) or not (ROM simulation)
+                if ((it_elem)->Has(HROM_WEIGHT))
+                    h_rom_simulation = true;

```

Comment:
this is a problem! all of the elements will write on the same time onto "h_rom_simulation", so the result will be unpredictable! (you cannot do this!)

Reply:
Now we are using an open mp reduction, if at least one element is found with an HROM_WEIGHT, then the memeber variable h_rom_simulation is set to true

Reply:
this is still wrong, h_rom_simulation is shared, so you cannot write to it without protection

Reply:
Is it possible are looking at the outdated version ?